src/features/ci-cd/models/release-additions.ts

```diff
@@ -1,28 +1,159 @@
-import type { AbstractSqlModel } from '@balena/abstract-sql-compiler';
+import type {
+	AbstractSqlModel,
+	CastNode,
+	ConcatenateNode,
+	ExistsNode,
+} from '@balena/abstract-sql-compiler';
 
 export const addToModel = (abstractSql: AbstractSqlModel) => {
+	const isFinal: ExistsNode = [
+		'Exists',
+		['ReferencedField', 'release', 'revision'],
+	];
+
 	abstractSql.tables['release'].fields.push({
 		fieldName: 'is final',
 		dataType: 'Boolean',
 		required: true,
-		computed: [
-			'Equals',
-			['ReferencedField', 'release', 'release type'],
-			['EmbeddedText', 'final'],
-		],
+		computed: isFinal,
 	});
 
+	const semverField: ConcatenateNode = [
+		'Concatenate',
+		['ReferencedField', 'release', 'semver major'],
+		['EmbeddedText', '.'],
+		['ReferencedField', 'release', 'semver minor'],
+		['EmbeddedText', '.'],
+		['ReferencedField', 'release', 'semver patch'],
+	];
+
 	abstractSql.tables['release'].fields.push({
 		fieldName: 'semver',
 		dataType: 'Short Text',
 		required: true,
+		computed: semverField,
+	});
+
+	const createdAtTimestamp: CastNode = [
+		'Cast',
+		[
+			'Floor',
+			[
+				'Multiply',
+				['Totalseconds', ['ReferencedField', 'release', 'created at']],
+				['Number', 1000],
+			],
+		],
+		'Text',
+	];
+
+	abstractSql.tables['release'].fields.push({
+		fieldName: 'version',
+		dataType: 'JSON',
+		required: true,
 		computed: [
 			'Concatenate',
+			['EmbeddedText', '{'],
+
+			['EmbeddedText', '"raw": "'],
+			semverField,
+			[
+				'Cast',
+				[
+					'Case',
+					[
+						'When',
+						['NotExists', ['ReferencedField', 'release', 'revision']],
+						['Concatenate', ['EmbeddedText', '-'], createdAtTimestamp],
+					],
+					[
+						'When',
+						[
+							'Equals',
+							['ReferencedField', 'release', 'revision'],
+							['Number', 0],
+						],
+						['EmbeddedText', ''],
+					],
+					[
+						'Else',
+						[
+							'Concatenate',
+							['EmbeddedText', '+rev'],
+							['ReferencedField', 'release', 'revision'],
+						],
+					],
+				],
+				'Text',
+			],
```

Comment:
I'm not sure how beneficial it really is but you could use:
```ts
function sqlConcat(
	[start, ...strings]: TemplateStringsArray,
	...nodes: Array<TextTypeNodes | string>
) {
	const concats: ConcatenateNode = ['Concatenate'];
	const addNode = (node: typeof nodes[number]) => {
		if (typeof node === 'string') {
			if (node.length > 0) {
				concats.push(['EmbeddedText', node]);
			}
		} else {
			concats.push(node);
		}
	};
	addNode(start);
	for (let i = 0; i < strings.length; i++) {
		addNode(nodes[i]);
		addNode(strings[i]);
	}

	return concats;
}
```
and do things like
```ts
sqlConcat`${['ReferencedField', 'release', 'semver major']}.${['ReferencedField', 'release', 'semver minor']}.${['ReferencedField', 'release', 'semver patch']}`
```
to get
```js
[
  'Concatenate',
  [ 'ReferencedField', 'release', 'semver major' ],
  [ 'EmbeddedText', '.' ],
  [ 'ReferencedField', 'release', 'semver minor' ],
  [ 'EmbeddedText', '.' ],
  [ 'ReferencedField', 'release', 'semver patch' ]
]

Reply:
@Page- I absolutely loved how the result looks like :heart: 
That's one of the JS features that I often forget.